functions/Get-DbaExternalProcess.ps1

```diff
@@ -0,0 +1,70 @@
+function Get-DbaExternalProcess {
+    <#
+    .SYNOPSIS
+        Gets OS processes created by SQL Server
+
+    .DESCRIPTION
+        Gets OS processes created by SQL Server
+
+        Helps when finding hung sessions with External Wait Types
+
+        https://web.archive.org/web/20201027122300/http://vickyharp.com/2013/12/killing-sessions-with-external-wait-types/
+
+    .PARAMETER ComputerName
+        The target SQL Server host computer
+
+    .PARAMETER Credential
+        Allows you to login to $ComputerName using alternative credentials.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Process
+        Author: Chrissy LeMaire (@cl), netnerds.net
+
+        Website: https://dbatools.io
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbaExternalProcess
+
+    .EXAMPLE
+        PS C:\> Get-DbaExternalProcess -ComputerName SERVER01, SERVER02
+
+        Gets OS processes created by SQL Server on SERVER01 and SERVER02
+
+    #>
+    [CmdletBinding()]
+    param (
+        [parameter(ValueFromPipeline, Mandatory)]
+        [DbaInstanceParameter[]]$ComputerName,
+        [PSCredential]$Credential,
+        [switch]$EnableException
+    )
+    process {
+        foreach ($computer in $computername) {
+            try {
+                $processes = Get-DbaCmObject -ComputerName $computer -Credential $Credential -ClassName win32_process | Where-Object ParentProcessId -eq (Get-DbaCmObject -ComputerName $computer -Credential $Credential -ClassName win32_process | Where-Object ProcessName -eq "sqlservr.exe").ProcessId | Select-DefaultView -Property 'PSComputer as ComputerName', ProcessId, Name, HandleCount, WorkingSetSize, VirtualSize
```

Comment:
Should $computer be resolved by Resolve-DbaNetworkName?

Reply:
Will the Where-Object execute the inner Get-DbaCmObject for each process comming over the pipeline? I would split this in more lines to be able to add verbose output to find problems.

Reply:
done-a-rama with the sep variables

Reply:
i'm not so sure about resolving the network name. not a fan today 😅